src/app/ui.cljs

```diff
@@ -4,26 +4,33 @@
 
 
 
-(defsc Backdrop [this {:ui/keys [modal?]}]
+(defsc Backdrop [this {:ui/keys [modal] :as props}]
   {}
-  (when modal?
-    (dom/div {:className "Backdrop" :onClick #(js/alert "backdrop")})))
+  (when (:ui/modal modal)
+    (dom/div {:className "Backdrop" :onClick
+                         #(comp/transact! this [(app.component.Burger.build-controls/toggle-modal props)])} "Less")))
 
 (def ui-backdrop (comp/factory Backdrop))
 
 
-(defsc Modal [this {:ui/keys [modal?]
+(defsc Modal [this {:ui/keys [modal]
                     :order/keys [burger] :as props}]
-  {:query [:ui/id :ui/modal? :order/burger]
+  {:query [:ui/id :ui/modal :order/burger]
    :ident (fn [] [:singleton ::modal])
    :initial-state {:ui/id 1
-                   :ui/modal? false}}
-  (js/console.log "Modal: " modal?)
-
-  (let [css (if modal? :.show :.noShow)]
-
-    (dom/div {:classes ["Modal" css]}
-      (ui-backdrop props))))
-
+                   :ui/modal false}}
+  (let [css (if (:ui/modal modal) :.show :.noShow)
+        burger-data (map #(select-keys % [:ingredient :count]) (:order/ingredients burger))]
```

Comment:
Waar komt `:order/ingredients` nu vandaag? Want ik zie hier geen join op.

Reply:
Die `:order/ingredients` komt uit de burger:
`{:order/id 1
    :order/total-price 4.0
    :order/ingredients [{:id 0 :ingredient "Bacon" :count 0 :price 0.5}
                                       {:id 1 :ingredient "Salad" :count 0 :price 0.1}
                                       {:id 2 :ingredient "Cheese" :count 0 :price 0.2}
                                       {:id 3 :ingredient "Meat" :count 1 :price 1.0}]}`

Reply:
Dat snap ik maar dat zou ik graat terug willen zien in de query..

Reply:
done